build-support/scripts/test-flake.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+pushd $(dirname ${BASH_SOURCE[0]}) > /dev/null
+SCRIPT_DIR=$(pwd)
+pushd ../.. > /dev/null
+SOURCE_DIR=$(pwd)
+pushd build-support/docker > /dev/null
+IMG_DIR=$(pwd)
+popd > /dev/null
+
+source "${SCRIPT_DIR}/functions.sh"
+
+IMAGE="travis-img-v0.13"
+CONTAINER="travis-cnt"
+GOOS="linux"
+GOARCH="amd64"
+TEST_BINARY="flake.test"
+
+function usage {
+cat <<-EOF
+Usage: test-flake [<options ...>]
+
+Description:
+
+  test-flake surfaces flakiness in tests by constraining CPU resources.
+  
+  Single or package-wide tests are run for multiple iterations with a configurable
+  amount of CPU resources. 
+
+  0.15 CPUs and 30 iterations are configured as sane defaults.
+
+  See Docker docs for more info on tuning 'cpus' param: 
+  https://docs.docker.com/config/containers/resource_constraints/#cpu
+
+Options:
+
+  --pkg=""             Target package
+  --test=""            Target test (requires pkg flag)
+  --cpus=0.15          Amount of CPU resources for container
+  --n=30               Number of times to run tests
+
+Examples:
+
+  ./test-flake.sh --pkg connect/proxy
+  ./test-flake.sh --pkg connect/proxy --cpus 0.20
+  ./test-flake.sh --pkg connect/proxy --test Listener
+  ./test-flake.sh --pkg connect/proxy --test TestUpstreamListener
+  ./test-flake.sh --pkg connect/proxy --test TestUpstreamListener -n 100
+EOF
+}
+
+function build_repro_env {
+    # Arguments:
+    #   $1 - pkg, Target package
+    #   $2 - test, Target tests
+    #   $3 - cpus, Amount of CPU resources for container
+    #   $4 - n, Number of times to run tests
+
+    APP=$(pwd | awk '{n=split($0, a, "/"); print a[n]}')
+
+    echo -e "App:\t\t$APP"
+    echo -e "Package:\t$1"
+    echo -e "Test:\t\t$2"
+    echo -e "CPUs:\t\t$3"
+    echo -e "Iterations:\t$4"
+    echo
+
+    echo "-----> Cleaning up old containers..."
```

Comment:
Not a huge deal at all but you could use the `status` or `status_stage` functions available and it will colorize the output for you.

It could help to distinguish this scripts output from a subcommands output.